Remove incorrect noexcept specifications from algorithms that may throw

timmiesmith opened the pull request:
Additional static analysis of oneDPL tests has identified other algorithms specified as 'noexcept' that may throw.  This PR extends similar work done in #1640.

mmichel11 reviewed (approved):
LGTM after CI passes

danhoeflinger commented on include/oneapi/dpl/pstl/algorithm_impl.h:
```diff
@@ -112,7 +112,7 @@ __brick_walk1(_ForwardIterator __first, _ForwardIterator __last, _Function __f,
 __brick_walk1(_RandomAccessIterator __first, _RandomAccessIterator __last, _Function __f,
-              /*vector=*/::std::true_type) noexcept
+              /*vector=*/::std::true_type)
 {
```
So is it the provided function that may throw, or just simply the bracket dereference operator? (within `__simd_walk_1`)

timmiesmith replied:
It's the body of the functor '__f'' that may throw.  It's several layers up in the stack from this point.

SergeyKopienko commented:
I think this problem i smore complicated.
Looks like this PR fixed only one problem case in code and probably we still have a lot of another cases.

timmiesmith commented:
> I think this problem i smore complicated. Looks like this PR fixed only one problem case in code and probably we still have a lot of another cases.

I'm addressing the issues as they're identified by the static analysis.  I'd prefer to continue this approach instead of taking a broader approach that may remove the qualifier where we may actually be able to use it.

SergeyKopienko reviewed (approved):
LGTM
